From fda613ba3a8a2b6e74f135cd4754b4665e5869fb Mon Sep 17 00:00:00 2001
From: nico <nicorg2529@gmail.com>
Date: Sun, 30 Dec 2018 09:02:28 +0000
Subject: [PATCH 3/5] TelephonyComponentFactory: Overload
 makeSubscriptionInfoUpdater

* Required for compatibility with pre-O-MR1 blobs

* Port from Change-Id: I2eaf297b4ccaede620c099d575a694ce63a32030 and Idb78221a1627d030c63e39312fb61f80adb1c782

Change-Id: I1b60de37e7ece16730afa3509b0f751c8f31aec9
---
 .../internal/telephony/TelephonyComponentFactory.java  | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/java/com/android/internal/telephony/TelephonyComponentFactory.java b/src/java/com/android/internal/telephony/TelephonyComponentFactory.java
index 604c1a61f..652adc5fa 100644
--- a/src/java/com/android/internal/telephony/TelephonyComponentFactory.java
+++ b/src/java/com/android/internal/telephony/TelephonyComponentFactory.java
@@ -225,8 +225,18 @@ public class TelephonyComponentFactory {
         return SubscriptionController.init(c, ci);
     }
 
+    public SubscriptionInfoUpdater makeSubscriptionInfoUpdater(Context context, Phone[] phones,
+            CommandsInterface[] ci) {
+        Rlog.d(LOG_TAG, "makeSubscriptionInfoUpdater - compat");
+        return null;
+    }
+
     public SubscriptionInfoUpdater makeSubscriptionInfoUpdater(Looper looper, Context context,
             Phone[] phones, CommandsInterface[] ci) {
+        SubscriptionInfoUpdater subInfoUpdater = makeSubscriptionInfoUpdater(context, phones, ci);
+        if (subInfoUpdater != null)
+            return subInfoUpdater;
+
         Rlog.d(LOG_TAG, "makeSubscriptionInfoUpdater");
         return new SubscriptionInfoUpdater(looper, context, phones, ci);
     }
-- 
2.17.1

